drivers/input/input_ite_it8801_kbd.c

```diff
@@ -0,0 +1,230 @@
+		/* Clear pending interrupts */
+		ret = i2c_reg_write_byte_dt(data->i2c_dev, config->reg_ksieer, GENMASK(7, 0));
+		/* Enable KSI falling edge event trigger interrupt */
+		ret = i2c_reg_write_byte_dt(data->i2c_dev, config->reg_ksiier, GENMASK(7, 0));
+	} else {
+		/* Disable KSI falling edge event trigger interrupt */
+		ret = i2c_reg_write_byte_dt(data->i2c_dev, config->reg_ksiier, 0x00);
+	}
+
+	if (ret != 0) {
+		LOG_ERR("Failed to set detect mode (ret %d)", ret);
+		return;
+	}
+}
+
+static int kbd_it8801_init(const struct device *dev)
+{
+	const struct kbd_it8801_config *const config = dev->config;
+	struct kbd_it8801_data *data = dev->data;
+	int ret;
+
+	/* Verify multi-function parent is ready */
+	if (!device_is_ready(config->mfd)) {
+		LOG_ERR("(input)%s is not ready", config->mfd->name);
+		return -ENODEV;
+	}
+
+	data->kbd_dev = dev;
+	data->i2c_dev = mfd_it8801_get_i2c_dt_spec(config->mfd);
+
+#if IT8801_MFDCTRL_COUNT
+	int status;
+
+	for (int i = 0; i < IT8801_MFDCTRL_COUNT; i++) {
+		/* Switching the pin to KSO alternate function (KSO[21:18]) */
+		status = mfd_it8801_configure_pins(data->i2c_dev, config->altctrl[i].gpiocr,
+						   config->altctrl[i].pin,
+						   config->altctrl[i].alt_func);
+		if (status != 0) {
+			LOG_ERR("Failed to configure KSO[21:18] pins");
+			return status;
+		}
+	}
+#endif
+	/* Disable wakeup and interrupt of KSI pins before configuring */
+	kbd_it8801_set_detect_mode(dev, false);
+
+	/* Start with KEYBOARD_COLUMN_ALL, KSO[22:11, 6:0] output low */
+	ret = i2c_reg_write_byte_dt(data->i2c_dev, config->reg_ksomcr, IT8801_REG_MASK_AKSOSC);
+	/* Gather KSI interrupt enable */
+	ret = i2c_reg_write_byte_dt(data->i2c_dev, IT8801_REG_GIECR, IT8801_REG_MASK_GKSIIE);
+	/* Alert response enable */
+	ret = i2c_reg_write_byte_dt(data->i2c_dev, IT8801_REG_SMBCR, IT8801_REG_MASK_ARE);
+	if (ret != 0) {
+		LOG_ERR("Failed to initialization setting (ret %d)", ret);
+		return ret;
+	}
+
+	return input_kbd_matrix_common_init(dev);
+}
+
+#if IT8801_MFDCTRL_COUNT
+static const struct it8801_mfd_input_altctrl_cfg it8801_input_altctrl[IT8801_MFDCTRL_COUNT] =
+	IT8801_DT_MFD_ITEMS_LIST(0);
+#endif
+
+INPUT_KBD_MATRIX_DT_INST_DEFINE(0);
+
+static const struct input_kbd_matrix_api kbd_it8801_api = {
+	.drive_column = kbd_it8801_drive_column,
+	.read_row = kbd_it8801_read_row,
+	.set_detect_mode = kbd_it8801_set_detect_mode,
+};
+
+static const struct kbd_it8801_config kbd_it8801_cfg_0 = {
+	.common = INPUT_KBD_MATRIX_DT_INST_COMMON_CONFIG_INIT(0, &kbd_it8801_api),
+	.mfd = DEVICE_DT_GET(DT_INST_PARENT(0)),
+#if IT8801_MFDCTRL_COUNT
+	.altctrl = it8801_input_altctrl,
+#endif
+	.kso_mapping = DT_INST_PROP(0, kso_mapping),
+	.reg_ksomcr = DT_INST_REG_ADDR_BY_IDX(0, 0),
+	.reg_ksidr = DT_INST_REG_ADDR_BY_IDX(0, 1),
+	.reg_ksieer = DT_INST_REG_ADDR_BY_IDX(0, 2),
+	.reg_ksiier = DT_INST_REG_ADDR_BY_IDX(0, 3),
+};
+
+static struct kbd_it8801_data kbd_it8801_data_0;
+
+PM_DEVICE_DT_INST_DEFINE(0, input_kbd_matrix_pm_action);
```

Comment:
Drivers should support multiple instances.

Reply:
Fixed, use multiple instances.